docs/src/tutorials/adding_new_problem_model.md

```diff
@@ -0,0 +1,85 @@
+Finally, developing a custom model decision model that will employ an optimization model under the hood requires understanding JuMP.jl.
+
+2. Need to employ [anonymous constraints and variables in JuMP](https://jump.dev/JuMP.jl/stable/manual/variables/#anonymous_variables)
+and register the constraints, variables and other optimization objects into PowerSimulations.jl's optimization container. Otherwise the
+features to use your problem in the simulation like the coordination with other problems and post processing won't work. More on this in the section [How to develop your `build_model!` function](@ref) below.
+
+3. Overload the required methods for your custom decision models. In some cases it will be possible to re-use some of the other methods that exist in PowerSimulations to make life easier for variable addition and constraint creation but this is not required.
+
+## Decision Problem
+
+### Step 1: Define a Custom Decision Problem
+
+Define a decision problem struct as a subtype of `PowerSimulations.DecisionProblem`. This requirement will enable a lot of the underlying functionality that relies on multiple dispatch. DecisionProblems are used to parameterize the behavior of `DecisionModel` objects which are just containers
+for the parameters, references and the optimization problem.
+
+It is possible to define a Custom Decision Problem that gives the user full control over the build, solve and execution process since it imposes less requirements on the developer. However, with less requirements there are also less check and validations performed inside of PowerSimulations which might lead to unexpected erros.
+
+```julia
+struct MyCustomDecisionProblem <: PSI.DecisionProblem end
+```
+
+Alternatevely, it is possible to define a Custom Decision Problem subtyping from `DefaultDecisionProblem` which imposes more requirements and structure onto the developer but employs more checks and validations in the process. Be aware that this route will decrease the flexibility of what can be done inside the custom model.
+
+```julia
+struct MyCustomDecisionProblem <: PSI.DefaultDecisionProblem end
+```
+
+Once the problem type is defined, initialize the decision model container with your custom decision problem passing the solver and some of the settings you need for the solution of the problem. For custom problems some of the settings need manual implementation by the developer. Settings availability is also dependent on wether  you choose to subtype from `PSI.DecisionProblem` or `PSI.DefaultDecisionProblem`
+
+```julia
+my_model = DecisionModel{MyCustomDecisionProblem}(
+    sys;
+    name = "MyModel",
+    optimizer = optimizer_with_attributes(HiGHS.Optimizer),
+     optimizer_solve_log_print = true,
+)
+```
+
+#### Mandatory Method Overloads
```

Comment:
I would use `overload` in cases where you add a method that replaces an existing method. In the case of these custom types, the user must implement (or extend) these methods because they don’t exist.